packages/shared/src/governance.ts

```diff
@@ -893,3 +893,60 @@ export function governanceE2ETests<
     })
   })
 }
+
+export async function treasurySpendForeignAssetTest(relayClient: NetworkClient, assetHubClient: NetworkClient) {
+  await relayClient.dev.setStorage({
+    System: {
+      account: [
+        // give Alice some DOTs so that she can sign a payout transaction.
+        [[devAccounts.alice.address], { providers: 1, data: { free: 10000e10 } }],
+      ],
+    },
+  })
+  const USDT_ID = 1984
+  const balanceBefore = await assetHubClient.api.query.assets.account(USDT_ID, devAccounts.alice.address)
+
+  // amount is encoded into the call
+  const amount = 123123123123n
+  const treasurySpendCall =
+    '0x130504000100a10f0002043205011f07b3c3b5aa1c0400010100d43593c715fdd31c61141abd04a99fd6822c8558854ccde39a5684e7a56da27d00'
+  await scheduleInlineCallWithOrigin(relayClient, treasurySpendCall, { system: 'Root' })
+  await relayClient.dev.newBlock()
+  await checkSystemEvents(relayClient, { section: 'treasury', method: 'AssetSpendApproved' })
+    // values (e.g. index) inside data increase over time,
+    // PET framework often rounds them.
+    // Tests will be flaky if we don't redact them.
+    .redact({ hash: false, redactKeys: /data/ })
+    .toMatchSnapshot('treasury spend approval events')
+
+  // filter events to find an index to payout
+  const spendEvents = (await relayClient.api.query.system.events()).filter(
+    ({ event }) => event.section === 'treasury' && event.method === 'AssetSpendApproved',
+  )
+  expect(spendEvents.length).toBe(1)
+  const spendEvent = spendEvents[0]
+  const spendIndex = (spendEvent.event.data[0] as any).toNumber()
```

Comment:
Great!

However, both here and below you can assert on the type of the event to get type information for free.
This is unlikely to happen, but should the order of the events' fields change, this will break, whereas if you `assert` and get the typecast for free, that won't be the case.

Here is my previous example on how to do this: https://github.com/open-web3-stack/polkadot-ecosystem-tests/blob/431ecfb37c559ddf7072b7edd85d57c17200bca4/packages/shared/src/proxy.ts#L1266-L1267